tests/CribblyBackend.UnitTests/ControllerTests/PlayerControllerTests.cs

```diff
@@ -104,59 +109,88 @@ public async Task GetByEmail_ShouldReturnPlayerAndOkStatus()
-            mockPlayerService.Setup(x => x.GetByEmail(req.Email)).ReturnsAsync(new Player() { });
+            var req = new LoginRequest();
+            AddTestUser(playerController, "auth id", "abc@abc.com");
+            mockPlayerService.Setup(x => x.ExistsAsync(It.IsAny<string>())).ReturnsAsync(false);
 
             var result = Assert.IsType<BadRequestObjectResult>(await playerController.Login(req));
+            mockPlayerService.VerifyAll();
         }
 
         [Fact]
         public async Task Login_ShouldCreatePlayerAndReturn200_IfPlayerDoesNotExist()
         {
             var req = new LoginRequest()
             {
-                Email = "abc@abc.com",
                 Name = "name"
             };
-            mockPlayerService.Setup(x => x.Exists(It.IsAny<string>())).ReturnsAsync(false);
-            mockPlayerService.Setup(x => x.Create(It.IsAny<string>(), It.IsAny<string>())).ReturnsAsync(new Player() { });
+            AddTestUser(playerController, "auth id", "abc@abc.com");
+            mockPlayerService.Setup(x => x.ExistsAsync(It.IsAny<string>())).ReturnsAsync(false);
+            mockPlayerService.Setup(x => x.CreateAsync(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>())).ReturnsAsync(new Player() { });
 
             var result = Assert.IsType<OkObjectResult>(await playerController.Login(req));
             var response = Assert.IsType<LoginResponse>(result.Value);
             Assert.False(response.IsReturning);
-            mockPlayerService.Verify(x => x.Create(req.Email, req.Name), Times.Once());
+            mockPlayerService.Verify(x => x.CreateAsync("auth id", "abc@abc.com", req.Name), Times.Once());
+        }
+
+
+        private void AddTestUser(ControllerBase controller, string authId, string email = "")
+        {
+            var user = NewIdentityUser(authId, email);
+            controller.ControllerContext = new ControllerContext();
+            controller.ControllerContext.HttpContext = new DefaultHttpContext { User = user };
+        }
+
+        private void AddHeader(ControllerBase controller, string header, string value)
+        {
+            var ctx = controller.ControllerContext.HttpContext;
+            if (controller.ControllerContext.HttpContext == null)
+            {
+                controller.ControllerContext.HttpContext = new DefaultHttpContext();
+            }
+            controller.ControllerContext.HttpContext.Request.Headers.Add(header, value);
+        }
+
+        private ClaimsPrincipal NewIdentityUser(string authProviderId, string email = "")
+        {
+            var claims = new List<Claim>
+            {
+                new(ClaimTypes.NameIdentifier, authProviderId),
+            };
+            if (!string.IsNullOrEmpty(email))
+            {
+                claims.Add(new(ClaimTypes.Email, email));
+            }
+            return new ClaimsPrincipal(new ClaimsIdentity(claims));
```

Comment:
These are all going to go away in a future PR and become more broadly applicable test helpers